23 add expense period

simo2898 opened the pull request.

GinoGreen committed:
new db

GinoGreen committed:
cdn dayjs formatting date

GinoGreen committed:
fix edit expence cause walletTransaction

GinoGreen committed:
fix crud expense cause walletTransaction

simo2898 committed:
Add Axios in package.json

simo2898 committed:
Adjust creation and modify of an expense

simo2898 committed:
Adjust duplication of an expense

simo2898 committed:
Adjust ExpensesView.vue

GinoGreen committed:
logs deleted

GinoGreen committed:
add col table

GinoGreen committed:
refactor table categories

GinoGreen committed:
refactor table secondary_categories

GinoGreen committed:
create new subcategory

GinoGreen committed:
new col

GinoGreen committed:
subcategory table fixed

GinoGreen committed:
fix

GinoGreen committed:
+ add
+ edit
+ delete
@Subcategory

GinoGreen committed:
CRUD Subcategory

GinoGreen committed:
new_db

GinoGreen committed:
fix error empty array

GinoGreen committed:
active link header

GinoGreen committed:
improvment form expenses

GinoGreen committed:
fix PUT expenses cause subcategory

simo2898 committed:
Add wallet's value

mattiapitossi committed:
Extract ExpenseService interface

mattiapitossi committed:
Format code

mattiapitossi committed:
Extract CategoryService interface

simo2898 committed:
Delete wallet transaction

simo2898 committed:
Delete wallet transaction

simo2898 committed:
Added wallet value

simo2898 committed:
Add Expense Period
Add disable wallet button when name is null or empty
Change some variable to private for encapsulation

simo2898 committed:
Add Expense Period Button and form

simo2898 committed:
Add WalletController Test
Change wallet method name from deleteCategory to deleteWallet
Delete extra classes

mattiapitossi committed:
Create dependabot.yml

mattiapitossi committed:
Create dependabot.yml

mattiapitossi committed:
Update dependabot.yml

mattiapitossi committed:
Update dependabot.yml

mattiapitossi committed:
Add npm dependency to dependabot

mattiapitossi committed:
Add npm dependency to dependabot

mattiapitossi committed:
Update dependabot.yml

mattiapitossi committed:
Update dependabot.yml

dependabot[bot] committed:
Bump spring-boot-starter-parent from 2.6.6 to 2.7.5

Bumps [spring-boot-starter-parent](https://github.com/spring-projects/spring-boot) from 2.6.6 to 2.7.5.
- [Release notes](https://github.com/spring-projects/spring-boot/releases)
- [Commits](https://github.com/spring-projects/spring-boot/compare/v2.6.6...v2.7.5)

---
updated-dependencies:
- dependency-name: org.springframework.boot:spring-boot-starter-parent
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump spring-boot-starter-parent from 2.6.6 to 2.7.5

Bumps [spring-boot-starter-parent](https://github.com/spring-projects/spring-boot) from 2.6.6 to 2.7.5.
- [Release notes](https://github.com/spring-projects/spring-boot/releases)
- [Commits](https://github.com/spring-projects/spring-boot/compare/v2.6.6...v2.7.5)

---
updated-dependencies:
- dependency-name: org.springframework.boot:spring-boot-starter-parent
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jackson-core from 2.13.2 to 2.14.0

Bumps [jackson-core](https://github.com/FasterXML/jackson-core) from 2.13.2 to 2.14.0.
- [Release notes](https://github.com/FasterXML/jackson-core/releases)
- [Commits](https://github.com/FasterXML/jackson-core/compare/jackson-core-2.13.2...jackson-core-2.14.0)

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson.core:jackson-core
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jackson-core from 2.13.2 to 2.14.0

Bumps [jackson-core](https://github.com/FasterXML/jackson-core) from 2.13.2 to 2.14.0.
- [Release notes](https://github.com/FasterXML/jackson-core/releases)
- [Commits](https://github.com/FasterXML/jackson-core/compare/jackson-core-2.13.2...jackson-core-2.14.0)

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson.core:jackson-core
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jackson-databind from 2.13.2.2 to 2.14.0

Bumps [jackson-databind](https://github.com/FasterXML/jackson) from 2.13.2.2 to 2.14.0.
- [Release notes](https://github.com/FasterXML/jackson/releases)
- [Commits](https://github.com/FasterXML/jackson/commits)

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson.core:jackson-databind
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jackson-databind from 2.13.2.2 to 2.14.0

Bumps [jackson-databind](https://github.com/FasterXML/jackson) from 2.13.2.2 to 2.14.0.
- [Release notes](https://github.com/FasterXML/jackson/releases)
- [Commits](https://github.com/FasterXML/jackson/commits)

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson.core:jackson-databind
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mattiapitossi committed:
Add docker-compose.yml for database creation

mattiapitossi committed:
Add docker-compose.yml for database creation

mattiapitossi committed:
Update readme

mattiapitossi committed:
Update readme

mattiapitossi committed:
Update README.md

mattiapitossi committed:
Update README.md

dependabot[bot] committed:
Bump jackson-databind from 2.14.0 to 2.14.1

Bumps [jackson-databind](https://github.com/FasterXML/jackson) from 2.14.0 to 2.14.1.
- [Release notes](https://github.com/FasterXML/jackson/releases)
- [Commits](https://github.com/FasterXML/jackson/commits)

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson.core:jackson-databind
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jackson-databind from 2.14.0 to 2.14.1

Bumps [jackson-databind](https://github.com/FasterXML/jackson) from 2.14.0 to 2.14.1.
- [Release notes](https://github.com/FasterXML/jackson/releases)
- [Commits](https://github.com/FasterXML/jackson/commits)

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson.core:jackson-databind
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jackson-core from 2.14.0 to 2.14.1

Bumps [jackson-core](https://github.com/FasterXML/jackson-core) from 2.14.0 to 2.14.1.
- [Release notes](https://github.com/FasterXML/jackson-core/releases)
- [Commits](https://github.com/FasterXML/jackson-core/compare/jackson-core-2.14.0...jackson-core-2.14.1)

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson.core:jackson-core
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jackson-core from 2.14.0 to 2.14.1

Bumps [jackson-core](https://github.com/FasterXML/jackson-core) from 2.14.0 to 2.14.1.
- [Release notes](https://github.com/FasterXML/jackson-core/releases)
- [Commits](https://github.com/FasterXML/jackson-core/compare/jackson-core-2.14.0...jackson-core-2.14.1)

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson.core:jackson-core
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mattiapitossi committed:
Create codeql.yml

mattiapitossi committed:
Create codeql.yml

mattiapitossi committed:
Create ethicalcheck.yml

mattiapitossi committed:
Create ethicalcheck.yml

mattiapitossi committed:
Update docker-compose for schema creation

mattiapitossi committed:
Update docker-compose for schema creation

mattiapitossi committed:
Update README

mattiapitossi committed:
Update README

simo2898 committed:
Merge branch '32-adding-a-new-item-in-wallet-wont-allow-you-to-insert-the-first-value' into 23-add-expense-period

simo2898 committed:
Added Recurring Bills Put, Delete and Post (For duplicating an item)
Added ajax calls for new endpoints in the recurring bill's component

simo2898 committed:
Added Recurring Bills Put, Delete and Post (For duplicating an item)
Added ajax calls for new endpoints in the recurring bill's component

simo2898 committed:
Added Recurring Bills Put, Delete and Post (For duplicating an item)
Added ajax calls for new endpoints in the recurring bill's component

simo2898 committed:
Merge remote-tracking branch 'origin/23-add-expense-period' into 23-add-expense-period

simo2898 committed:
Added job that create expenses from recurring bills when the application is started

simo2898 committed:
Added job that create expenses from recurring bills when the application is started

simo2898 committed:
Added methods that create an expense from an recurring bills, and update the recurring's next payment

simo2898 committed:
Added methods that create an expense from an recurring bills, and update the recurring's next payment

simo2898 committed:
Merge remote-tracking branch 'origin/23-add-expense-period' into 23-add-expense-period

simo2898 committed:
Merge branch '32-adding-a-new-item-in-wallet-wont-allow-you-to-insert-the-first-value' into 23-add-expense-period

# Conflicts:
#	src/main/java/com/expensetracker/controller/ExpensePeriodController.java

mattiapitossi commented on src/main/java/com/expensetracker/cronjobs/ExpensesCronJobs.java:
```diff
@@ -0,0 +1,25 @@
+
+@RequiredArgsConstructor
+@Component
+public class ExpensesCronJobs {
```
Should we change this to a service?

mattiapitossi commented on src/main/java/com/expensetracker/mapper/impl/ExpenseMapperImpl.java:
```diff
@@ -0,0 +1,29 @@
+import com.expensetracker.model.ExpensePeriod;
+import org.springframework.stereotype.Component;
+
+@Component
```
There should be an annotation called @Mapper

simo2898 replied:
The annotation comes from a dependency called org.mapstruct

mattiapitossi commented on src/main/java/com/expensetracker/service/impl/ExpensePeriodServiceImpl.java:
```diff
@@ -0,0 +1,132 @@
+        LocalDate nextPayment = null;
+
+        switch (expensePeriod.getPeriodType()) {
+            case "YEAR":
```
You could also use the enhanced Switch/Case from Java 17 so you don't need to use the break.

mattiapitossi commented on src/main/java/com/expensetracker/service/impl/ExpensePeriodServiceImpl.java:
```diff
@@ -0,0 +1,132 @@
+            case "DAY":
+                nextPayment = expensePeriod.getNextPayment().plusDays(expensePeriod.getPeriodDate());
+                break;
+        }
```
Default?

simo2898 committed:
Changed switch case to java 17 switch case
Removed sout from ExpensePeriodController
Removed unused repository from WalletController
Changed ExpensesCronJob name to ExpensesUtilsService and added Service annotation

simo2898 committed:
Changed nextPayment assignment to the switch

mattiapitossi committed:
Merge branch '20-migrate-from-react-to-vue' into 23-add-expense-period
